Add createAlarm action

jakebathman opened the pull request:
**Checks**

- [x] I've checked there are no linting errors.
- [x] I've checked the code is still at 100% test coverage.

**Added Actions (if relevant)**

- createAlarm

**Are you happy to be listed as a contributor on [Shortcuts.fun](https://shortcuts.fun)?**

✅ Yes

**Any other information / comments**

I really appreciated the contributing guide, but did have some confusion going through the steps. I've got some notes on that and I'll make a separate PR for some updates to contributing.md. Otherwise it seemed to go alright, assuming the changes look good to you!

joshfarrant commented on src/interfaces/WF/WFWorkflowActionParameters.ts:
```diff
@@ -61,6 +61,7 @@ interface WFWorkflowActionParameters {
   WFDontIncludeFileExtension?: boolean;
   WFEncodeMode?: WFEncodeMode;
   WFFlashlightSetting?: WFFlashlightSetting;
+  WFFrequency?: string[];
```
Could this be changed to `DayOfWeek[]` please? 🙂

regaw-leinad replied:
The frequency can also be the `Ask When Run`, and I believe the structure is

```json
"WFItemSpecifier": {
  "Value": {
    "Type": "Ask"
  },
  "WFSerializationType": "WFTextTokenAttachment"
}
```

There's a iOS client bug where this value isn't persisted when set when using `Ask When Run` in the create alarm action.  Try setting the frequency to `Ask When Run` in app, Clicking `Done`, then going back into the shortcut editor. It will be reset to `Never`.

This should still be implemented at some point, but more thought will have to go into how to handle how many different ways `Ask When Run` is used (list item, inline variable, magic variable, etc)

joshfarrant replied:
After the discussion in the other comment it will now need to be changed to:

```ts
WFFrequency?: WFFrequency[];
```

joshfarrant commented on src/interfaces/WF/DayOfWeek.ts:
```diff
@@ -0,0 +1,11 @@
+type DayOfWeek = (
```
I was debating this too. It's hard to say without knowing where else (if anywhere) this type will be used.  

> I [...] went with DayOfWeek to hopefully reduce future confusion and make it interoperable with others that might need a similar param.

Totally agree with this, and this was my thought process when I reviewed this PR too. However, after a bit more thought, I'm leaning towards going with `WFFrequency` instead. If we do find other actions which accept the day of the week via a different property name then we can switch it back out to `DayOfWeek` and have `WFFrequency` and others essentially just alias `DayOfWeek`. That way we keep things consistent but also DRY.

Would you be happy switching it to `WFFrequency` for now @jakebathman?

jakebathman commented:
Alrighty @joshfarrant, I addressed the feedback and rebased it on the latest `master` to make things easier.

If you'd like to test it (or anyone else), here's what I used: https://gist.github.com/jakebathman/b4e35565b29d11f9cf2bcf1ffc0f346d

Thanks!
